tests/functional/upgrade/test_non_stretch_cluster_upgrade.py

```
@@ -0,0 +1,529 @@
+    8.  Verify no IO is happening on the stopped node.
+    9.  Verify workloads migrated to a healthy node automatically.
+    10. Confirm data integrity (md5sum comparison for both old and new workloads).
+    11. Recover the node; remove the taint; run IO on migrated pods.
+    12. Delete migrated pods; verify Deployment recreates them.
+    13. No data loss/corruption after recovery.
+    14. Deploy fresh workloads pinned to the recovered node; run IOs.
+    """
+
+    _tainted_node = []
+
+    def _cleanup_taints():
```

Ruff (0.15.20)

[warning] Missing return type annotation for private function `_cleanup_taints`

Add return type annotation: `None`

(ANN202)